### PR TITLE
Add additional parameters to ms build runner

### PR DESCRIPTION
**Pull request by bradtwurst**
Provides the ability to add
* /noconsolelogger (/noconlog)
* /detailedsummary (/ds)

~~and a 'catch-all' for any other parameters (AddParameters(string))~~

### CODE REVIEW

**Comment by patriksvensson** [comment]
@bradtwurst Great! I don't think the catch-all should be included though, since this normally is done via the `ArgumentCustomization` delegate on the `ToolSettings` class. We want to be consistent about this.

**Comment by bradtwurst** [comment]
no problem regarding catch-all.  I didn't realize that existed.  I'll update the pull request

**Comment by gep13** [comment]
@bradtwurst in line with our contributing process (https://github.com/cake-build/cake/blob/develop/CONTRIBUTING.md) can you please create an issue, and link to this PR from it?

Also, would it be possible for you to rebase your changes on top of the latest develop branch? Thanks!

File: src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs [hunk 1]
```diff
@@ -58,6 +58,18 @@ private ProcessArgumentBuilder GetArguments(FilePath solution, MSBuildSettings s
                 builder.Append(settings.MaxCpuCount > 0 ? string.Concat("/m:", settings.MaxCpuCount) : "/m");
             }
 
+            // Set the detailed summary flag
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs]: Sentence should end with a period.

File: src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs [hunk 2]
```diff
@@ -58,6 +58,18 @@ private ProcessArgumentBuilder GetArguments(FilePath solution, MSBuildSettings s
...
+            if (settings.DetailedSummary.GetValueOrDefault())
+            {
+                builder.Append("/ds");
+            }
+
+            // Set the no console logger flag
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSBuild/MSBuildRunner.cs]: Sentence should end with a period.

File: src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs [hunk 3]
```diff
@@ -70,6 +70,21 @@ public sealed class MSBuildSettings : ToolSettings
         public bool? NodeReuse { get; set; }
 
         /// <summary>
+        /// Gets or sets whether or not detailed summary is created
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs]: Sentence should end with a period.

File: src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs [hunk 4]
```diff
@@ -70,6 +70,21 @@ public sealed class MSBuildSettings : ToolSettings
...
+        ///  Shows detailed information at the end of the build
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs]: Remove indentation.

File: src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs [hunk 5]
```diff
@@ -70,6 +70,21 @@ public sealed class MSBuildSettings : ToolSettings
...
+        ///  about the configurations built and how they were
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs]: Remove indentation.

File: src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs [hunk 6]
```diff
@@ -70,6 +70,21 @@ public sealed class MSBuildSettings : ToolSettings
...
+        ///  scheduled to nodes.
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs]: Remove indentation.

File: src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs [hunk 7]
```diff
@@ -70,6 +70,21 @@ public sealed class MSBuildSettings : ToolSettings
...
+        /// </summary>
+        public bool? DetailedSummary { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether or not information is logged to the console
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/MSBuild/MSBuildSettings.cs]: Sentence should end with a period.

**Comment by gep13** [comment]
relates to #1312

**Comment by gep13** [comment]
@bradtwurst I have rebased and merged this PR, and also added the PR Review comments from @patriksvensson.  You can see this merged in here: https://github.com/cake-build/cake/commit/e9e91f0e55014cef5420123eee10d4b1760cc12d

As a result, I am going to manually close this PR, but your contribution has been pulled in, and correctly attributed.  Thank you for taking the time to contribute to cake! 👍

**Pull request closed without merging**